pkg/client/unversioned/helper.go

```diff
@@ -269,7 +269,9 @@ func NegotiateVersion(client *Client, c *Config, version string, clientRegistere
 	}
 	apiVersions, err := client.ServerAPIVersions()
 	if err != nil {
-		return "", fmt.Errorf("couldn't read version from server: %v", err)
+		// This is almost always a connection error, and higher level code should treat this as a generic error,
+		// not a version specific error.
```

Comment:
"version specific" is ambiguous. I guess you mean "negotiation specific"? Could you reword?